pipeline_utils/lib/yaml_parser.py

```diff
@@ -0,0 +1,397 @@
+#!/usr/bin/env python
+
+import yaml
+
+
+###############################################################
+#   Functions
+###############################################################
+def load_yaml(file):
+    """
+        return a generator to loaded yaml documents in file
+    """
+    with open(file) as stream:
+        try:
+            for d in yaml.safe_load_all(stream):
+                yield d
+        except yaml.YAMLError as exc:
+            sys.exit(exc)
+
+
+###############################################################
+#   YamlWfl, Yaml Workflow
+###############################################################
+class YamlWfl(object):
+
+    def __init__(self, yaml_d):
+        """
+        """
+        for key, val in yaml_d.items():
+            setattr(self, key, val)
+        self._validate()
+
+    def _validate(self):
+        """
+        """
+        try:
+            getattr(self, 'name')
+            getattr(self, 'description')
+            getattr(self, 'runner')
+            getattr(self, 'input')
+            getattr(self, 'output')
+        except AttributeError as e:
+            raise ValueError('JSON validation error, {0}\n'
```

Comment:
These aren't JSON, so consider changing
```suggestion
            raise ValueError('Validation error, {0}\n'
```

Reply:
changed